### PR TITLE
[Enhancement] Add TASK_SOURCE column to information_schema.task_runs

### PR DESCRIPTION
**Pull request by Youngwb**
## Why I'm doing:

`information_schema.task_runs` exposes one row per task run but not which subsystem submitted it, so an MV refresh run can't be told apart from a CTAS / INSERT / PIPE / DATACACHE_SELECT run without cross-referencing other tables.

## What I'm doing:

Add a `TASK_SOURCE` column to `information_schema.task_runs`, sourced from the task's `Constants.TaskSource`: `MV` / `CTAS` / `INSERT` / `PIPE` / `DATACACHE_SELECT`, or `UNKNOWN` for legacy rows whose source was never recorded. Covers the FE system table, the `TTaskRunInfo` thrift struct, the BE schema scanner, plus FE UT and docs (en/zh/ja).

Fixes #issue: N/A — observability enhancement, no tracking issue.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
  - [ ] This pr needs auto generate documentation
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 4.1
  - [ ] 4.0
  - [ ] 3.5

### CODE REVIEW

**Comment by CelerData-Reviewer** [comment]
@codex review

**Comment by CelerData-Reviewer** [comment]
@codex review

**Comment by Youngwb** [comment]
@codex review

File: fe/fe-core/src/main/java/com/starrocks/catalog/system/information/TaskRunsSystemTable.java [hunk 1]
```diff
@@ -235,6 +236,7 @@ public static TGetTaskRunInfoResult query(TGetTasksParams params) {
             info.setProperties(status.getPropertiesJson());
             info.setProcess_time(status.getProcessStartTime() / 1000);
             info.setJob_id(status.getStartTaskRunId());
+            info.setTask_source(status.getSource() == null ? "UNKNOWN" : status.getSource().name());
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/system/information/TaskRunsSystemTable.java]: Good catch — this was a real bug, though the root cause is one layer up. This line was already correct; the problem was that `status.getSource()` could never actually return `null`. `TaskRunStatus.source` was initialized to `Constants.TaskSource.CTAS`, and Gson leaves an absent field at its Java default, so a status persisted before the `source` field existed deserialized with `source == CTAS` and surfaced here as `CTAS` instead of `UNKNOWN`.

Fixed by removing that default so legacy rows deserialize to `source == null`, which this line then maps to `UNKNOWN`. Removing the default also exposed a latent NPE in `TaskRunStatus.getLastRefreshState()` (`!source.isMVTask()` on a null source), now a null-safe `source != Constants.TaskSource.MV`. Added regression tests in `TaskRunStatusTest` (legacy JSON → null source; null source → no NPE).

File: test/sql/test_information_schema/T/test_task_run_status [hunk 2]
```diff
@@ -20,6 +20,22 @@ SELECT count(1) FROM information_schema.materialized_views WHERE TABLE_SCHEMA =
 [UC]task_name=SELECT TASK_NAME FROM information_schema.materialized_views WHERE TABLE_SCHEMA = 'db_${uuid0}' AND TABLE_NAME='mv1';
 SELECT count(1) FROM information_schema.task_runs WHERE TASK_NAME = '${task_name}';
 SELECT count(1) FROM information_schema.task_runs WHERE task_name = '${task_name}';
+SELECT TASK_SOURCE FROM information_schema.task_runs WHERE task_name = '${task_name}';
+
+submit task task_src_insert_${uuid0} as insert into ss select * from ss;
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply test/sql/test_information_schema/T/test_task_run_status:25]: Thanks — the interaction is real (the new INSERT/CTAS runs do appear in `task_runs`), but these `select count(*)` lines can't fail on it because they're prefixed `[UC]`. The SQL-Tester framework records `[UC]` results but does not compare them: the comparison step runs only under `if not record_mode and not uncheck` in `test/lib/sr_sql_lib.py`, and `[UC]` sets `uncheck = True`, so the recorded `7` is never asserted.

The real assertion beside each is `select if (count(*) > 0, "OK", "FAILED")` → `OK`, a `> 0` invariant that holds regardless of how many runs exist. Every count in this test that is actually compared is scoped with `WHERE task_name = ...` / `WHERE query_id = ...` to a specific name, so the new (differently-named) tasks don't change those results. No update needed here.

**Review by kangkaisen — DISMISSED** [review]

**Comment by HangyuanLiu** [comment]
I think we should refine the semantics of `TaskRunStatus.source` here.

Changing the default from `CTAS` to no default does avoid mislabeling legacy task runs as `CTAS`. However, making `private Constants.TaskSource source;` implicitly nullable also leaves a fairly easy footgun for future code. The current call sites are mostly null-safe, for example `TaskRunsSystemTable` maps null to `UNKNOWN`, and `getLastRefreshState()` avoids the NPE path. But a future caller can still naturally write `status.getSource().isMVTask()` or `status.getSource().name()` and hit an NPE.

I think it would be better to add an explicit `UNKNOWN` sentinel to `Constants.TaskSource`, and use it as the legacy `TaskRunStatus` value instead of relying on null:

```java
public enum TaskSource {
    UNKNOWN,
    CTAS,
    MV,
    INSERT,
    PIPE,
    DATACACHE_SELECT;
}
```

Then `TaskRunStatus` can either default to `UNKNOWN`, or at least make `getSource()` normalize null to `UNKNOWN`. This also makes the `information_schema.task_runs` output consistent with the internal model and makes future call sites less error-prone.

One more minor point: the comment saying legacy runs are “never MV refreshes” is not quite accurate. A legacy run might have been an MV refresh; we just do not have a reliable persisted source in `TaskRunStatus`. I would phrase it as: legacy statuses without a recorded source are displayed as `UNKNOWN` and use the non-MV fallback path for refresh-state handling.

**Comment by Youngwb** [comment]
Thanks @HangyuanLiu — good points, adopted in 6363284.

**1 & 2 (UNKNOWN sentinel):** Added an explicit `UNKNOWN` to `Constants.TaskSource` and made `TaskRunStatus.source` default to it, so `getSource()` is never null and the internal model matches the `information_schema.task_runs` output. That removes the nullable footgun entirely: `getLastRefreshState()` reverts to the original `!source.isMVTask()` (UNKNOWN is non-MV, no NPE), and `TaskRunsSystemTable` simplifies to `status.getSource().name()`.

Verified the enum addition is safe: `TaskSource` has no `switch` / `ordinal()` / `values()[]` usage, and Gson serializes enums by name, so existing persisted `"CTAS"`/`"MV"` values are unaffected.

**3 (comment accuracy):** You're right — "never MV refreshes" was inaccurate; a legacy run may well have been an MV refresh, we just have no recorded source. Removed that comment (the code is back to the upstream form) and reworded the field comment to say legacy runs have no recorded source and surface as UNKNOWN.

Also updated the regression tests to assert `UNKNOWN` instead of null.

**Review by HangyuanLiu — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]
